asynction/mock_server.py

```diff
@@ -240,6 +245,17 @@ def _register_handlers(
                 with_bindings_validation = bindings_validator_factory(channel.bindings)
                 connect_handler = with_bindings_validation(connect_handler)
 
+            if server_security:
+                if not self.spec.components.security_schemes:
+                    raise UnregisteredSecurityScheme
```

Comment:
Based on the comments I made under `types.py` this validation here is redundant. Essentially it can be moved under the `AsyncAPISpec` class.